lib/Db/TagMapper.php

```diff
@@ -134,33 +134,37 @@ public function getAllTagsForMessages(array $messages, string $userId): array {
 			return $message->getMessageId();
 		}, $messages);
 
+		$tags = [];
 		$qb = $this->db->getQueryBuilder();
 		$tagsQuery = $qb->selectDistinct(['t.*', 'mt.imap_message_id'])
 			->from($this->getTableName(), 't')
 			->join('t', 'mail_message_tags', 'mt', $qb->expr()->eq('t.id', 'mt.tag_id', IQueryBuilder::PARAM_INT))
 			->where(
-				$qb->expr()->in('mt.imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY)),
+				$qb->expr()->in('mt.imap_message_id', $qb->createParameter('ids')),
 				$qb->expr()->eq('t.user_id', $qb->createNamedParameter($userId, IQueryBuilder::PARAM_STR))
 			);
-		$queryResult = $tagsQuery->execute();
 
-		$tags = [];
-		while (($row = $queryResult->fetch()) !== false) {
-			$messageId = $row['imap_message_id'];
-			if (!isset($tags[$messageId])) {
-				$tags[$messageId] = [];
-			}
+		foreach (array_chunk($ids, 1000) as $chunk) {
+			$tagsQuery->setParameter('ids', $chunk, IQueryBuilder::PARAM_INT_ARRAY);
+			$queryResult = $tagsQuery->execute();
+
+			while (($row = $queryResult->fetch()) !== false) {
```

Comment:
this done doesn't actually work. `fetch` won't return any rows but always `false`, despite `$queryResult->rowCount()` returning 1. Investigating.